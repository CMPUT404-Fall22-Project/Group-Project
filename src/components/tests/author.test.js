import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import Author from "../author";

it("snapshot test", () => {
  const component = renderer.create(
    <Author
      displayName="testName"
      author_id="testID"
      github="https://github.com/"
      profileImage="https://avatars.githubusercontent.com/u/56713153?v=4"
      id="currentID"
    ></Author>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("check that data is correct", () => {
  const { getByLabelText, getByText } = render(
    <Author
      displayName="testName"
      author_id="testID"
      github="https://github.com/"
      profileImage="https://avatars.githubusercontent.com/u/56713153?v=4"
    />
  );
  const displayNameTest = getByText("testName").textContent;
  expect(displayNameTest).toEqual("testName");

  const idTest = getByText("testID").textContent;
  expect(idTest).toEqual("testID");

  expect(getByLabelText("GitHub").hasAttribute("href", "https://github.com"));
});
