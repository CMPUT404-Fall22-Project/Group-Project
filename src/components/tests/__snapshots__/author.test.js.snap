// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
      >
        <img
          alt="Profile Picture"
          className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="https://avatars.githubusercontent.com/u/56713153?v=4"
        />
      </div>
    </div>
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
      >
        testName
      </span>
      <span
        className="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
      >
        testID
      </span>
    </div>
  </div>
  <div
    className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
  >
    <button
      aria-label="Follow"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="PersonAddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </button>
    <a
      aria-label="GitHub"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      href="https://github.com/"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="GitHubIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
        />
      </svg>
    </a>
  </div>
</div>
`;
